### PR TITLE
add saml and oidc example

### PR DESCRIPTION
**Pull request by ykethan**
#### Description of changes:

adds SAML and OIDC to auth concepts 

#### Related GitHub issue #, if available:
closes: https://github.com/aws-amplify/docs/issues/7560
closes: https://github.com/aws-amplify/docs/issues/7561

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/auth/concepts/external-identity-providers/index.mdx [hunk 1]
```diff
@@ -342,6 +342,93 @@ When you import and use the `signInWithRedirect` function, it will add a listene
...
+    },
+  },
+});
+```
+
+### Set up your frontend
+
+<InlineFilter filters={["angular", "javascript", "nextjs", "react", "vue"]}>
+
+Use the `signInWithRedirect` API to initiate sign-in with an SAML identity provider.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/concepts/external-identity-providers/index.mdx:421]:
```suggestion
Use the `signInWithRedirect` API to initiate sign-in with a SAML identity provider.
```

File: src/pages/[platform]/build-a-backend/auth/concepts/external-identity-providers/index.mdx [hunk 2]
```diff
@@ -342,6 +342,93 @@ When you import and use the `signInWithRedirect` function, it will add a listene
...
+
+await signInWithRedirect({
+  custom: 'MicrosoftEntraID'
+});
+```
+</InlineFilter>
+
+## Configure SAML provider
+
+To setup a SAML provider, you can configure them in `amplify/auth/resource.ts` file on the defineAuth handler. For example, if you would like to setup a Microsoft EntraID provider, you can do so as follows:
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/concepts/external-identity-providers/index.mdx:391]:
```suggestion
To setup a SAML provider, you can configure them in your `amplify/auth/resource.ts` file. For example, if you would like to setup a Microsoft EntraID provider, you can do so as follows:
```

File: src/pages/[platform]/build-a-backend/auth/concepts/external-identity-providers/index.mdx [hunk 3]
```diff
@@ -342,6 +342,93 @@ When you import and use the `signInWithRedirect` function, it will add a listene
 </Accordion>
 </InlineFilter>
 
+## Configure OIDC provider
+
+To setup a OIDC provider, you can configure them in `amplify/auth/resource.ts` file on the defineAuth handler. For example, if you would like to setup a Microsoft EntraID provider, you can do so as follows:
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/concepts/external-identity-providers/index.mdx:347]:
```suggestion
To setup a OIDC provider, you can configure them in your `amplify/auth/resource.ts` file. For example, if you would like to setup a Microsoft EntraID provider, you can do so as follows:
```

File: src/pages/[platform]/build-a-backend/auth/concepts/external-identity-providers/index.mdx [hunk 4]
```diff
@@ -342,6 +342,93 @@ When you import and use the `signInWithRedirect` function, it will add a listene
...
+        'https://mywebsite.com/profile',
+      ],
+    },
+  },
+});
+```
+
+### Set up your frontend
+
+<InlineFilter filters={["angular", "javascript", "nextjs", "react", "vue"]}>
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/concepts/external-identity-providers/index.mdx:376]: can we link out to the sign-in page? these duplicate headings are going to cause issues with the anchor, and the inline filter will show an empty section

Comment (ykethan) [reply src/pages/[platform]/build-a-backend/auth/concepts/external-identity-providers/index.mdx:376]: https://github.com/aws-amplify/docs/pull/7566/files#diff-d59fe95d17fcc94afb4be025157ed7e88a6b2fb97f15755da89c6127e1b19c53L347 
but we do have it at already at end of page

can change to eg: setup SAML redirect on frontend

**Review by josefaidt — APPROVED** [review]

File: src/pages/[platform]/build-a-backend/auth/concepts/external-identity-providers/index.mdx [hunk 5]
```diff
@@ -342,6 +342,89 @@ When you import and use the `signInWithRedirect` function, it will add a listene
...
+
+Use the `signInWithRedirect` API to initiate sign-in with an OIDC identity provider.
+
+```ts title="src/my-client-side-js.js"
+import { signInWithRedirect } from 'aws-amplify/auth';
+
+await signInWithRedirect({
+  custom: 'MicrosoftEntraID'
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/auth/concepts/external-identity-providers/index.mdx:382]:
```suggestion
  provider: {
    custom: 'MicrosoftEntraID'
  }
```

File: src/pages/[platform]/build-a-backend/auth/concepts/external-identity-providers/index.mdx [hunk 6]
```diff
@@ -342,6 +342,91 @@ When you import and use the `signInWithRedirect` function, it will add a listene
...
+
+<InlineFilter filters={["angular", "javascript", "nextjs", "react", "vue"]}>
+
+Use the `signInWithRedirect` API to initiate sign-in with a SAML identity provider.
+
+```ts title="src/my-client-side-js.js"
+import { signInWithRedirect } from 'aws-amplify/auth';
+
+await signInWithRedirect({
+  custom: 'MicrosoftEntraIDSAML'
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/auth/concepts/external-identity-providers/index.mdx:425]:
```suggestion
  provider: {
    custom: 'MicrosoftEntraID'
  }
```

**Review by renebrandel — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**